src/git/Index.cpp

```diff
@@ -422,15 +422,14 @@ Id Index::indexId(const QString &path, uint32_t *mode) const
 
 Id Index::workdirId(const QString &path, uint32_t *mode) const
 {
-  Repository repo(git_index_owner(d->index));
+  const git_index_entry *entry = git_index_get_bypath(d->index, path.toUtf8(), GIT_INDEX_ENTRY_STAGEMASK);
```

Comment:
Consider calling `Index::entry()` here, depending on whether you think that future edits/improvements to `Index::entry()` would have to be reflected here too (and could be missed by accident).